src/snowflake/connector/storage_client.py

```diff
@@ -159,17 +159,19 @@ def encrypt(self) -> None:
                 meta.real_src_file_name,
                 tmp_dir=self.tmp_dir,
             )
-            meta.upload_size = os.path.getsize(self.data_file)
+            meta.upload_size = os.path.getsize(meta.real_src_file_name)
         else:
             encrypted_stream = BytesIO()
             src_stream = meta.src_stream or meta.intermediate_stream
             src_stream.seek(0)
             self.encryption_metadata = SnowflakeEncryptionUtil.encrypt_stream(
                 meta.encryption_material, src_stream, encrypted_stream
             )
-            src_stream.seek(0)
-            meta.upload_size = encrypted_stream.seek(0, os.SEEK_END)
+            # Reset streams
             encrypted_stream.seek(0)
+            src_stream.seek(0)
+            meta.upload_size = src_stream.seek(0, os.SEEK_END)
```

Comment:
Is this correct that we are not using `encrypted_stream` for upload size?

Reply:
Yup, see my PR description. I should have a detailed explanation in there. I verified that old connector versions did this

Reply:
But is that the desired behavior? We explicitly add this padding for stored proc so that the returned result is the padded size. Encrypted size is consistent with the `ls @stage` result for the file 🤔